src/libmongoc/src/mongoc/mongoc-crypt.c

```diff
@@ -2001,6 +2001,12 @@ _mongoc_crypt_rewrap_many_datakey (_mongoc_crypt_t *crypt,
    mongocrypt_binary_t *filter_bin = NULL;
    bool ret = false;
 
+   if (master_key) {
+      // Caller must verify that `provider` is not-NULL when `master_key` is
+      // not-NULL.
+      BSON_ASSERT_PARAM (provider);
+   }
+
```

Comment:
```suggestion
   // Caller must ensure `provider` is provided alongside `master_key`.
   BSON_ASSERT (!master_key || provider);

```
Small simplification.